pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -718,6 +724,91 @@ func (r *ReconcilePerconaServerMongoDB) deleteMongosIfNeeded(cr *api.PerconaServ
 	return r.deleteMongos(cr)
 }
 
+func (r *ReconcilePerconaServerMongoDB) reconcileConfigMap(cr *api.PerconaServerMongoDB, repls []*api.ReplsetSpec) error {
+	for _, rs := range repls {
+		configMap := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "ConfigMap",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      psmdb.MongodConfigCMName(cr.Name, rs.Name),
+				Namespace: cr.Namespace,
+			},
+			Data: map[string]string{
+				"mongod.conf": rs.Configuration,
+			},
+		}
+
+		err := setControllerReference(cr, configMap, r.scheme)
+		if err != nil {
+			return errors.Wrap(err, "failed to set controller ref for config map")
+		}
+
+		if rs.Configuration != "" {
+			err = createOrUpdateConfigmap(r.client, configMap)
+			if err != nil {
+				return errors.Wrap(err, "failed to create/update mongod config map")
+			}
+		} else {
+			err = deleteConfigmapIfExists(r.client, cr, configMap.Namespace, configMap.Name)
+			if err != nil {
+				return errors.Wrap(err, "failed to delete mongod config map")
+			}
+		}
+	}
+
+	return nil
+}
+
+func deleteConfigmapIfExists(cl client.Client, cr *api.PerconaServerMongoDB, nsName, cmName string) error {
+	var configMap = &corev1.ConfigMap{}
+
+	err := cl.Get(context.TODO(), types.NamespacedName{
+		Namespace: nsName,
+		Name:      cmName,
+	}, configMap)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrap(err, "get config map")
+	}
+
+	if k8serrors.IsNotFound(err) {
+		return nil
+	}
+
+	if !metav1.IsControlledBy(configMap, cr) {
+		return nil
+	}
+
+	err = cl.Delete(context.Background(), configMap)
+	if err != nil {
+		return errors.Wrap(err, "delete config map")
```

Comment:
we can return `cl.Delete` because on upper level we already wrapping this error as configmap deletion errror

Reply:
changed in 885c4a7